frontend/src/TasksDashboard.jsx

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+import { loadTasks } from './data/tasks/actions';
+import { loadNodes } from './data/nodes/actions';
+import { isErrored, isLoaded, isLoading, isNotLoaded } from './data/utils';
+import Table from './Table';
+
+const mapStateToProps = ({ tasks, nodes, clusters }) => ({
+    tasks,
+    nodes,
+    clusters,
+});
+
+const mapDispatchToProps = {
+    loadTasks,
+    loadNodes,
+}
+
+class TasksDashboard extends React.Component {
+    static propTypes = {
+        tasks: PropTypes.shape({
+            loadingState: PropTypes.oneOf(['NOT_LOADED', 'LOADING', 'LOADED', 'ERROR']).isRequired,
+            data: PropTypes.object,
+            error: PropTypes.object,
+        }).isRequired,
```

Comment:
Feels like this shape could be refactored

Reply:
Agreed! https://github.com/EDITD/ElasticSurgery/pull/18/commits/13d9e67cccf837531db619baa90cc71b1c79e110